knowledge_runtime/knowledge_runtime/middleware/auth.py

```
@@ -0,0 +1,57 @@
+
+from fastapi import Depends, HTTPException, status
+from fastapi.security import HTTPAuthorizationCredentials, HTTPBearer
+
+from knowledge_runtime.config import get_settings
+
+# HTTPBearer security scheme for OpenAPI documentation
+security = HTTPBearer(auto_error=False)
+
+
+def verify_internal_token(
+    credentials: Optional[HTTPAuthorizationCredentials] = Depends(security),
```

Ruff (0.15.10)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)